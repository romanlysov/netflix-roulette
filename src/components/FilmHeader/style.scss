.header {
  max-width: 75rem;
  margin: 0 auto;
  padding-bottom: 2.81rem;
  background-image: linear-gradient(rgba(0, 0, 0, .8), rgba(0, 0, 0, .5)), url('../../../assets/images/netflix.jpg');
  background-repeat: no-repeat;
  background-size: cover;
}

.info-block {
  &__cover {
    width: 22.7rem;
  }
}

//.search-result > .info-block {
//  display: none;
//}
