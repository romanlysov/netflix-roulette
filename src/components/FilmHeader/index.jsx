import React from 'react'

import { StaticHeader } from '../StaticHeader'
import { MovieCard } from '../MovieCard'
import { FormContainer } from '../../containers/FormContainer'

import './style.scss'

export const FilmHeader = ({ film, className, onclick, searchButtonClass }) => {
  return (
    <header className={className}>
      <StaticHeader
        className="navigation"
        onclick={onclick}
        searchButtonClass={searchButtonClass}
      />
      <MovieCard film={film} />
      <FormContainer />
    </header>
  )
}
