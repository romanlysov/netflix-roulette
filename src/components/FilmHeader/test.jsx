import React from 'react'
import renderer from 'react-test-renderer'

import { Header } from 'components/FilmHeader/index'

jest.mock('../../containers/FormContainer')

const filmMock = {}

describe('FilmHeader', () => {
  test('renders FilmHeader for searchResults properly', () => {
    const wrapper = renderer.create(<Header />).toJSON()
    expect(wrapper).toMatchSnapshot()
  })
  test('renders header for MovieCard properly', () => {
    const wrapper = renderer.create(<Header film={filmMock} />).toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})
