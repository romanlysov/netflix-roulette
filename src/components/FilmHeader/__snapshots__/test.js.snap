// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header for searchResults properly 1`] = `
<header>
  <nav
    className="header__static navigation"
  >
    <img
      alt=""
      className="navigation__logo"
      src="test-file-stub"
    />
    <button
      className="navigation__search"
    >
      search
    </button>
  </nav>
</header>
`;

exports[`Header renders header for MovieCard properly 1`] = `
<header>
  <nav
    className="header__static navigation"
  >
    <img
      alt=""
      className="navigation__logo"
      src="test-file-stub"
    />
    <button
      className="navigation__search"
    >
      search
    </button>
  </nav>
  <div
    className="header__movie-info info-block"
  >
    <img
      alt=""
      className="info-block__cover"
    />
    <div
      className="info-block__description"
    >
      <h1
        className="info-block__title"
      />
      <span
        className="info-block__genre"
      />
      <div
        className="info-block__sub-info"
      >
        <span
          className="info-block__year"
        />
        <span
          className="info-block__length"
        >
           min
        </span>
      </div>
      <p
        className="info-block__annotation"
      />
    </div>
  </div>
</header>
`;
